main.py

```diff
@@ -87,10 +87,19 @@ def train(chunk_dir, save_file, load_file=None, generation_num=0,
 
     print("Training from:", tf_records[0], "to", tf_records[-1])
 
-    n = dual_net.DualNetworkTrainer(save_file)
+    n = dual_net.DualNetworkTrainer(save_file, logdir=logdir)
     with timer("Training"):
         n.train(tf_records, init_from=load_file,
-                logdir=logdir, num_steps=num_steps, verbosity=verbosity)
+                num_steps=num_steps, verbosity=verbosity)
+
+
+def validate(holdout_dir, load_file=None, logdir=None, num_steps=1000):
+    tf_records = sorted(gfile.Glob(os.path.join(holdout_dir, '*.tfrecord.zz')))
+    preprocessing.SHUFFLE_BUFFER_SIZE = int(1e5)
```

Comment:
this is strange. Why are you overriding a CONSTANT? Should be passed in if it's gonna be overridden.

Reply:
obsolete, removed. Thanks!